components/gateway/src/main/kotlin/net/corda/p2p/gateway/messaging/http/ReconfigurableHttpServer.kt

```diff
@@ -62,55 +53,53 @@ internal class ReconfigurableHttpServer(
             resources: ResourcesHolder,
         ): CompletableFuture<Unit> {
             val configUpdateResult = CompletableFuture<Unit>()
+            resources.keep {
+                httpServers.values.forEach {
+                    it.close()
+                }
+                httpServers.clear()
+            }
             @Suppress("TooGenericExceptionCaught")
             try {
-                if (newConfiguration.hostPort == oldConfiguration?.hostPort) {
-                    logger.info(
-                        "New server configuration for ${dominoTile.coordinatorName} on the same port, " +
-                            "HTTP server will have to go down"
-                    )
-                    serverLock.write {
-                        val oldServer = httpServer
-                        httpServer = null
+                val newServersConfiguration = newConfiguration.servers
+                if (newServersConfiguration.isEmpty()) {
+                    throw CordaRuntimeException("No servers defined!")
```

Comment:
```suggestion
                    throw IllegalArgumentException("No server configurations defined!")
```
I believe `CordaRuntimeException` is a type supposed to be used by flows, so not sure it should be used here.

Also, given the schema you have on corda-api, I understand this is just for the sake of defensive programming. But, empty lists will practically be rejected at the point the user specifies the configuration and it won't reach that point at all. Let me know if not, because ideally that should be the way it works.

Reply:
The JSON scheme should prevent the user from having an empty list. But, because it's in another library, I think having another level of protection here is better.

Reply:
yes, I am happy with having an extra layer of protection here. As long as, we know an empty list will be rejected at the early point as well now.

But, I think we should move away from `CordaRuntimeException` here for the reason described above.